webapp/resources/templates/fragments/loginProviders.html

```diff
@@ -22,7 +22,7 @@ <h3 th:utext="#{screen.welcome.label.loginwith}"/>
                     <span th:case="linkedin" class="fa fa-linkedin"></span>
                     <span th:case="facebook" class="fa fa-facebook-square"></span>
                     <span th:case="oidc" class="fa fa-openid"></span>
-                    <span th:case="saml2" class="fa fa-lock"></span>
+                    <span th:case="saml2" th:class="${entry.cssClass}" ></span>
```

Comment:
Pfft.  Thymeleaf's #strings.replace is a wrapper around Java's String.replace, and there doesn't appear to be an equivalent for String.replaceAll, so the tradeoff amounts to this:

- A simpler, Thymeleaf-only patch that either only escapes spaces or gets really ugly really quick (unless there's some clever trick I'm missing); or
- A more complex Java patch (like this one), which can escape the \W character class, and perhaps leave the door open to different strategies for mapping the provider name to a CSS class.

I suppose patching Thymeleaf to add a #strings.replaceAll() is an option, which I'm open to doing, but that will take a bit of time, and would end up pushing this down on my priority list.

What do you think?